crates/interpreter/src/instructions/context.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{
+    interpreter_types::{Jumps, LoopControl},
+    Host, Interpreter, InterpreterTypes,
+};
+
+use super::Instruction;
+
+pub struct InstructionContext<'a, H: Host + ?Sized, ITy: InterpreterTypes> {
```

Comment:
Should we need to announce the trait bound for `H` here?  This trait bound led to conduction to other declares. be like:
```rust
pub type Instruction<W, H> = fn(&mut context::InstructionContext<'_, H, W>);
```
Every types contain `Instruction` need to add trait bound `H: Host`, but don't required by `InstructionContext` itself.

Reply:
```diff
pub trait InstructionProvider {
    /// Context type.
-	type Context;
+	type Context: Host;
```

Reply:
Probably it is not needed,

Reply:
It is not needed indeed, thank you.